File: server/php/src/Application/Domain/Payments/PaymentDetails/AcquirerSelectionInformation.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class AcquirerSelectionInformation
{
    public ?int $fallbackLevel = null;
    public ?string $result = null;
    public ?string $ruleName = null;

    public function toArray(): array
    {
        return [
            'fallbackLevel' => $this->fallbackLevel,
            'result' => $this->result,
            'ruleName' => $this->ruleName,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/acquirer_selection_information.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class AcquirerSelectionInformation
          attr_accessor :fallback_level, :result, :rule_name

          def initialize(fallback_level: nil, result: nil, rule_name: nil)
            @fallback_level = fallback_level
            @result = result
            @rule_name = rule_name
          end

          def to_h
            {
              'fallbackLevel' => @fallback_level,
              'result' => @result,
              'ruleName' => @rule_name
            }
          end
        end
      end
    end
  end
end